torch/lib/c10d/ProcessGroupMPI.cpp

```diff
@@ -94,13 +94,17 @@ void checkSameSizeAndType(
 
 } // namespace
 
+std::vector<at::Tensor> ProcessGroupMPI::WorkMPI::result() {
+  return outputTensors_;
+}
+
 ProcessGroupMPI::AsyncWork::AsyncWork(
```

Comment:
Could you please add a paragraph to describe how tensor lifetime and ownership is managed among 1) user code 2) AsyncWork 3) WorkMPI 4) WorkEntry? Besides that, it will be super helpful to use send/recv and allreduce as two examples.

Reply:
I don't understand: tensors lifetime and ownership is handled by tensor library ref counting. we do no introduce anything new here?

for vectors: is it still relevant? right now each of AsyncWork, WorkMPI, WorkEntry keeps its own copy of user provided outputs.